src/librustdoc/html/highlight.rs

```diff
@@ -136,14 +158,23 @@ impl Iterator for TokenIter<'a> {
     }
 }
 
-fn get_real_ident_class(text: &str, edition: Edition) -> Class {
-    match text {
+/// Returns `None` if this is a `Class::Ident`.
+fn get_real_ident_class(text: &str, edition: Edition, allow_path_keywords: bool) -> Option<Class> {
+    let ignore: &[&str] =
+        if allow_path_keywords { &["self", "Self", "super", "crate"] } else { &["self", "Self"] };
+    if ignore.iter().any(|k| *k == text) {
+        return None;
+    }
+    Some(match text {
         "ref" | "mut" => Class::RefKeyWord,
-        "self" | "Self" => Class::Self_,
         "false" | "true" => Class::Bool,
         _ if Symbol::intern(text).is_reserved(|| edition) => Class::KeyWord,
-        _ => Class::Ident,
-    }
+        _ => return None,
+    })
+}
+
+fn move_span(file_span_lo: u32, start: u32, end: u32) -> (u32, u32) {
```

Comment:
Please document this

Reply:
I'll also rename the function. For the history about this function, originally it was receiving a span as argument (the file span which has the "lo" where the file begins) and then simply adds the byte pos to it so then we could retrieve the "definition span" from the "span map". In the end, I realized a span was too much as a key so I simply kept a pair of `u32`, which is smaller and simpler to handle.

Anyway, I'll update the function name as I told and also rename it.